content/4-component-composition/5-context/angular/user.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from "@angular/core";
+
+@Injectable()
+export class UserService {
+  user = {
+    id: 1,
+    username: "unicorn42",
+    email: "unicorn42@example.com",
+  };
+
+  updateUsername(username: string) {
+    this.user.username = username;
```

Comment:
Once again, I completely agree but it is mutated in every other demos of other frameworks/libs.
Since the goal is to compare them, I won't change it (plus it's the old version, the "bad" one :wink:)